sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,135 @@
-import {
-  AppBar,
-  AppBarProps,
-  Backdrop,
-  IconButton,
-  paperClasses,
-  Popover,
-  PopoverProps,
-  Typography
-} from '@mui/material';
-import CloseRounded from '@mui/icons-material/CloseRounded';
+import { AppBarProps, paperClasses, Popover, PopoverProps, useTheme } from '@mui/material';
 import ChatGPT, { ChatGPTProps } from './ChatGPT.tsx';
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
+import MinimizedBar from '@craftercms/studio-ui/components/MinimizedBar';
+import DialogHeader from '@craftercms/studio-ui/components/DialogHeader/DialogHeader';
+import AlertDialog from '@craftercms/studio-ui/components/AlertDialog';
+import PrimaryButton from '@craftercms/studio-ui/components/PrimaryButton/PrimaryButton';
+import SecondaryButton from '@craftercms/studio-ui/components/SecondaryButton/SecondaryButton';
+import { useSelector } from 'react-redux';
+import { GlobalState } from '@craftercms/studio-ui';
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
+  const theme = useTheme();
+  const dialogsState = useSelector((state: GlobalState) => state.dialogs);
   const {
     open,
     onClose,
-    appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const [openAlertDialog, setOpenAlertDialog] = useState(false);
+
+  const handleMinimize = useCallback(() => {
+    onMinimize?.();
+  }, [onMinimize]);
```

Comment:
Is this necessary? Why not call the prop directly? If it is, let's better add it to a ref (useUpdateRefs) so it doesn't get constantly recreated every render of the parent component didn't memo the callback.

Reply:
We don't need this anymore. Thanks for pointing out this.